### PR TITLE
Library Update

### PR DESCRIPTION
**Pull request by StriderDM**
Crate should be ready for another release.

Changes:
Moved get_flags from RandomXVM to RandomXFlag and renamed it to get_recommended_flags, this was a bug.
Finished last TODO in library, adding logic according to RandomX header where needed.
Fixed some minor formatting.
Updated tests.
Updated version.
Fixed formatting in documentation.

Note: Breaking changes exist in this PR

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -71,6 +71,23 @@ bitflags! {
...
+impl RandomXFlag {
+    /// Returns the recommended flags to be used.
+    ///
+    /// Does not include:
+    /// * FLAG_LARGE_PAGES
+    /// * FLAG_FULL_MEM
+    /// * FLAG_SECURE
+    ///
+    /// The above flags need to be set manually, if required.
+    pub fn get_recommended_flags() -> RandomXFlag {
```
Comment (SWvheerden) [inline src/lib.rs:83]: Should this not rather be named default?

Comment (StriderDM) [reply src/lib.rs:83]: Default is all flags not set (FLAG_DEFAULT which maps to RANDOMX_FLAG_DEFAULT).

Comment (StriderDM) [reply src/lib.rs:83]: So while this fn will return valid flags, it won't necessarily return FLAG_DEFAULT.

**Review by CjS77 — APPROVED** [review]
LGTM

**Pull request merged**